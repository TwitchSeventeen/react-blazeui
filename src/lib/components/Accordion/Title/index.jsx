import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

class Title extends React.Component {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }

  /**
   * Pass props to onClick when button is clicked
   * @param event
   */
  handleClick(event) {
    const { onClick } = this.props;

    onClick(event, this.props);
  }

  render() {
    const {
      children, className, active, dataName, index, ...other
    } = this.props;
    const classes = classNames('c-card__control', className);
    return (
      <button
        {...other}
        type="button"
        aria-expanded={active.toString()}
        className={classes}
        onClick={this.handleClick}
        index={index}
        data-name={dataName}
      >
        {children}
      </button>
    );
  }
}

Title.propTypes = {
  /**
   * Whether this is the active item
   */
  active: PropTypes.bool,
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * The index of the title in the accordion
   */
  index: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  /**
   * Function called when button is clicked
   */
  onClick: PropTypes.func,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Title.defaultProps = {
  children: null,
  className: null,
  active: false,
  index: null,
  onClick: null,
  dataName: 'accordion-title',
};

export default Title;
