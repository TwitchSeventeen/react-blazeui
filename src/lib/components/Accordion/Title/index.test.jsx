import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { shallow } from 'enzyme';
import Title from './index';

describe('Title', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Title).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
        <Title active onClick={jest.fn()}>
          lorem ipsum dolor
        </Title>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  /**
   * handleClick
   */
  it('can handle handleClick', () => {
    const onClickProp = jest.fn();
    const wrapper = shallow(<Title active onClick={onClickProp}/>);

    wrapper.instance().handleClick();

    expect(onClickProp).toHaveBeenCalled();
  });
});

