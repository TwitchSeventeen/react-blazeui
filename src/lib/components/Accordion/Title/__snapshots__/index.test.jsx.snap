// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders correctly 1`] = `
<DocumentFragment>
  <button
    aria-expanded="true"
    class="c-card__control"
    type="button"
  >
    lorem ipsum dolor
  </button>
</DocumentFragment>
`;
